internal/controller/transform.go

```diff
@@ -0,0 +1,137 @@
+package controller
+
+import (
+	"fmt"
+
+	"github.com/thanos-community/thanos-operator/api/v1alpha1"
+	"github.com/thanos-community/thanos-operator/internal/pkg/manifests"
+	manifestquery "github.com/thanos-community/thanos-operator/internal/pkg/manifests/query"
+	manifestqueryfrontend "github.com/thanos-community/thanos-operator/internal/pkg/manifests/queryfrontend"
+	manifestruler "github.com/thanos-community/thanos-operator/internal/pkg/manifests/ruler"
+	manifestsstore "github.com/thanos-community/thanos-operator/internal/pkg/manifests/store"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func queryV1Alpha1ToOptions(in v1alpha1.ThanosQuery) manifestquery.Options {
+	labels := manifests.MergeLabels(in.GetLabels(), in.Spec.Labels)
+	name := QueryNameFromParent(in.GetName())
+	opts := commonToOpts(name, in.GetNamespace(), in.Spec.Replicas, labels, in.Spec.CommonThanosFields, in.Spec.Additional)
+	return manifestquery.Options{
+		Options:       opts,
+		ReplicaLabels: in.Spec.QuerierReplicaLabels,
+		Timeout:       "15m",
+		LookbackDelta: "5m",
+		MaxConcurrent: 20,
+	}
+}
+
+// QueryNameFromParent returns the name of the Thanos Query component.
+func QueryNameFromParent(resourceName string) string {
+	return fmt.Sprintf("thanos-query-%s", resourceName)
```

Comment:
Should we check length and DNS name here?

Reply:
Yep, I commented in the description:

> We will likely still want to add validation to the names but at least it is clear where and what is happening now.

We can add that after. I want to check how k8s is generating such names in cases for statefulsets pods etc.